fineract-provider/src/main/java/org/apache/fineract/accounting/glaccount/domain/TrialBalance.java

```diff
@@ -100,8 +100,9 @@ public boolean equals(Object obj) {
         }
         TrialBalance other = (TrialBalance) obj;
         return Objects.equals(other.officeId, officeId) && Objects.equals(other.glAccountId, glAccountId)
-                && Objects.equals(other.amount, amount) && Objects.equals(other.entryDate, entryDate)
-                && Objects.equals(other.transactionDate, transactionDate) && Objects.equals(other.closingBalance, closingBalance);
+                && Objects.equals(other.amount, amount) && other.entryDate.compareTo(entryDate) == 0 ? Boolean.TRUE
+                        : Boolean.FALSE && other.transactionDate.compareTo(transactionDate) == 0 ? Boolean.TRUE
```

Comment:
Just curious about this:
How is `other.entryDate.compareTo(entryDate) == 0 ? Boolean.TRUE: Boolean.FALSE`
dIfferent from just  `other.entryDate.compareTo(entryDate) == 0 `  (In terms of end result) of the boolean?
Is it because .equals returns Boolean.TRUE and not primitive data-type boolean?

Reply:
compareTo returns an int. If values to be compared are equal, the value returned is zero.

Reply:
Agreed, so if other.entryDate.compareTo(entryDate) == 0 is executed, then 0 == 0 returns true, which is boolean. 
Sorry if I am confusing, I am not able to understand the need for the
 ```?Boolean.TRUE: Boolean.FALSE ``` part.

Reply:
This is just like a Java If statement where other.entryDate.compareTo(entryDate) == 0 is the condition, the first value Boolean.TRUE is return if the condition is true and the second value Boolean.FALSE if it is false. Could have used an If statement but this is shorter. See [this](https://www.baeldung.com/java-ternary-operator)

Reply:
@percyashu what @thesmallstar is pointing out here, and I agree with, is that `? Boolean.TRUE: Boolean.FALSE` is not required - as he said, just `other.entryDate.compareTo(entryDate) == 0` yields exactly the same value. The difference between primitive VS non-primitive is hidden by "auto boxing". Do you want to raise a follow-up PR to clean this up, or ignore this?